pkg/kubelet/kubelet_pods.go

```diff
@@ -1691,8 +1694,7 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 				ContainerID: cid,
 			}
 
-		case cs.State == kubecontainer.ContainerStateUnknown &&
-			oldStatus != nil && // we have an old status
+		case oldStatus != nil &&		// we have an old status
```

Comment:
I *think* this should fix it. Previously, this logic only kicked in when the current state is unknown, which caused default to always go to Waiting, even if there was an old status. I think that's the bug.